pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -90,70 +94,159 @@ func ctrRun(domainName, xenCfgFilename, imageHash string, envList map[string]str
 // but instead use containerd to set it up, and xl to run it, so this does not need to do anything.
 // If/when we invert it, and have containerd launch the container with a microvm wrapper,
 // *then* this will need to work.
-func ctrStop(id string, force bool) error {
-	log.Infof("ctrStop %s %t\n", id, force)
+func ctrStop(containerID string, force bool) error {
+	log.Infof("ctrStop %s %t\n", containerID, force)
 	log.Infof("ctr stop done\n")
 	return nil
 }
 
 // ctrRm remove an existing container
-func ctrRm(id string) error {
-	log.Infof("ctrRm %s\n", id)
+func ctrRm(containerID string) error {
+	log.Infof("ctrRm %s\n", containerID)
+	ctrdClient, err := getContainerdClient(ctrdSocket)
+	if err != nil {
+		log.Errorf("Could not build containerd client. %v", err.Error())
+		return fmt.Errorf("Could not build containerd client. %v", err.Error())
+	}
+	defer ctrdClient.Close()
+	ctrdCtx := getContainerdContext(CtrdServicesNamespace)
+
+	container, err := loadContainer(containerID)
+	if err != nil {
+		return err
+	}
+	task, err := container.Task(ctrdCtx, cio.NewAttach(cio.WithStdio))
```

Comment:
This could cause issues, since it uses a fifo, and only one attach is allowed at a time. See [here](https://godoc.org/github.com/containerd/containerd/cio#Attach):

> There should only be one reader for a task's IO set because fifo's can only be read from one reader or the output
> will be sent only to the first reads

Can you pass it `nil`? We aren't looking at the output here, so who cares?

Reply:
As per discussion with @rvs, I've removed create task step, instead will me manually mounting container's snapshot.

Reply:
Got it. What about the `cio.NewAttach()`?